### DISCUSSION TITLE
player.teleport will not teleport

### DISCUSSION THREAD

**Discussion by Xyphyn** [Minestom/Minestom#1980] [Help] [answered]
I have it so that when a game ends in a different instance, it sends them back to the lobby instance. In the lobby instance, I have this event:

instance.eventNode().addListener(AddEntityToInstanceEvent::class.java) {
            val player = it.entity as? Player

            if (player != null) {
                player.setGameMode(GameMode.ADVENTURE)
                player.inventory.clear()
                player.teleport(spawn)
            }
}

The gamemode is correctly set to adventure, the inventory is cleared, but they are not teleported to spawn. Why is this? There is no error message or anything.

**Comment by GreatWyrm** [comment, marked as answer]
I would advise against teleporting someone as they are joining the instance, and you should use: [Player#setInstance(instance, Point)](https://javadoc.minestom.net/net/minestom/server/entity/Entity.html#setInstance(net.minestom.server.instance.Instance,net.minestom.server.coordinate.Point)) instead.

That will work a lot better.
